consensus/istanbul/backend/engine.go

```diff
@@ -415,10 +363,8 @@ func (sb *Backend) UpdateValSetDiff(chain consensus.ChainReader, header *types.H
 	// If this is the last block of the epoch, then get the validator set diff, to save into the header
 	log.Trace("Called UpdateValSetDiff", "number", header.Number.Uint64(), "epoch", sb.config.Epoch)
 	if istanbul.IsLastBlockOfEpoch(header.Number.Uint64(), sb.config.Epoch) {
-		newValSet, err := validators.GetValidatorSet(header, state)
-		if err != nil {
-			log.Error("Istanbul.Finalize - Error in retrieving the validator set. Using the previous epoch's validator set", "err", err)
-		} else {
+		newValSet, err := sb.getNewValidatorSet(header, state)
+		if err == nil {
```

Comment:
Why remove the error message is err is non nil?  Also, seems like if there ever was an error from here: 
 https://github.com/celo-org/celo-blockchain/pull/524/files#diff-6e58dd6edf43343da874fb58a00f6727R368, it would be silently ignored.

Also, should this return the error if it's non nil?